tfjs-backend-webgpu/src/kernels/Bincount.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {Bincount, BincountAttrs, BincountInputs, KernelConfig, KernelFunc, TensorInfo, util} from '@tensorflow/tfjs-core';
+
+import {WebGPUBackend} from '../backend_webgpu';
+import {BincountProgram} from '../bincount_webgpu';
+
+import {fill} from './Fill';
+import {slice} from './Slice';
+
+export function bincount(
+    args:
+        {inputs: BincountInputs, backend: WebGPUBackend, attrs: BincountAttrs}):
+    TensorInfo {
+  const {inputs, backend, attrs} = args;
+  const {x, weights} = inputs;
+  const {size} = attrs;
+
+  const xSize = util.sizeFromShape(x.shape);
+  const weightsSize = util.sizeFromShape(weights.shape);
+  const hasWeights = weightsSize > 0;
+  const outputSize: [number] = [Math.max(xSize, size)];
```

Comment:
Please don't use the same `outputSize` for `BincountProgram` and `fill`. So you don't need to do extra `slice`.

Reply:
Using `slice` aims to dispatch threads according output shapes, which is the normal case in almost all operators.
If dispatching threads according input shapes, it can indeed get better performance. @haoyunfeix , had you mentioned that there was an error for `matmul_splitK` when using input shapes to dispatch threads?
I personally prefer to using output shapes here. @gyagp @hujiajie any comments here?

Reply:
Had changed dispatch size based on input size.